ext/standard/streamsfuncs.c

```diff
@@ -830,11 +827,16 @@ PHP_FUNCTION(stream_select)
 		} else if (usec < 0) {
 			zend_argument_value_error(5, "must be greater than or equal to 0");
 			RETURN_THROWS();
+		} else if (usec >= 1000000) {
+			php_error_docref(NULL, E_WARNING, "must be less than 1000000");
+			if (UNEXPECTED(EG(exception))) {
+				RETURN_THROWS();
+			}
```

Comment:
Can't this just be a `zend_argument_value_error`? We've shipped improved parameter validation all the time without an RFC and here the `<0` case already throws.

Reply:
I don't know if people rely on passing a large int or not for the microseconds